pandas/core/dtypes/dtypes.py

```diff
@@ -126,16 +131,20 @@ class PandasExtensionDtype(_DtypeOpsMixin):
     isnative = 0
     _cache = {}  # type: Dict[str_type, 'PandasExtensionDtype']
 
-    def __unicode__(self):
+    def __unicode__(self) -> str_type:
         return self.name
 
-    def __str__(self):
+    @property
```

Comment:
What error was this causing?

Reply:
It was an ``AttributeError``, caused by the call to .name in ``def __unicode__``

Reply:
Do subclasses assign `name` as a property or a class variable? If not the former then doing it this way will cause typing issue in the subclasses